### ISSUE TITLE
Incorrect type narrowing between deriving protocols

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pyright#5901]
**Describe the bug**

See sample code.

`P0` is a runtime-checkable protocol. `P1` is also a runtime-checkable protocol and it inherits `P0`. Variable `x` is typed as `P0`. When `x` is type-narrowed as `P1`, its type is incorrectly recognized as `Never` instead of `P1`.

Only occurs when `P1` is a protocol, but not when `P1` is a normal class.

**Code or Screenshots**

```python
from typing import Protocol, runtime_checkable

@runtime_checkable
class P0(Protocol):
    def f0(self, /) -> None:
        ...

@runtime_checkable
class P1(P0, Protocol):
    def f1(self, /) -> None:
        ...

def f(x: P0, /):
    if isinstance(x, P1):
        reveal_type(x)  # "Never"!!!
```

![image](https://github.com/microsoft/pyright/assets/15389744/fe5ce714-2a3c-4538-9ec9-76cabfbc7245)

**VS Code extension or command-line**

Pyright 1.1.326

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.327, which I just published. It will also be included in an upcoming release of pylance.
